make not working with boost 1.68

Environment: Ubuntu 16.04
When I'm running `make`, I get the following output:
 [19%] Building CXX object libraries/fc/tests/CMakeFiles/all_tests.dir/all_tests.cpp.o
In file included from /root/opt/boost_1_67_0/include/boost/test/unit_test_log.hpp:25:0,
                 from /root/opt/boost_1_67_0/include/boost/test/tools/old/impl.hpp:19,
                 from /root/opt/boost_1_67_0/include/boost/test/test_tools.hpp:46,
                 from /root/opt/boost_1_67_0/include/boost/test/unit_test.hpp:18,
                 from /root/steem/steem/libraries/fc/tests/all_tests.cpp:2:
/root/opt/boost_1_67_0/include/boost/test/unit_test_log.hpp:227:28: error: ‘boost::unit_test::{anonymous}::unit_test_log’ defined but not used [-Werror=unused-variable]
 BOOST_TEST_SINGLETON_INST( unit_test_log )
                            ^
/root/opt/boost_1_67_0/include/boost/test/utils/trivial_singleton.hpp:67:36: note: in definition of macro ‘BOOST_TEST_SINGLETON_INST’
 namespace { BOOST_JOIN( inst, _t)& inst = BOOST_JOIN( inst, _t)::instance(); }
                                    ^
cc1plus: all warnings being treated as errors
libraries/fc/tests/CMakeFiles/all_tests.dir/build.make:62: recipe for target 'libraries/fc/tests/CMakeFiles/all_tests.dir/all_tests.cpp.o' failed
make[2]: *** [libraries/fc/tests/CMakeFiles/all_tests.dir/all_tests.cpp.o] Error 1
CMakeFiles/Makefile2:420: recipe for target 'libraries/fc/tests/CMakeFiles/all_tests.dir/all' failed
make[1]: *** [libraries/fc/tests/CMakeFiles/all_tests.dir/all] Error 2
Makefile:127: recipe for target 'all' failed
make: *** [all] Error 2

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -253,8 +253,8 @@ if(ENABLE_COVERAGE_TESTING)
 endif()
 
 # fc/src/compress/miniz.c breaks strict aliasing. The Linux kernel builds with no strict aliasing
-SET(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -fno-strict-aliasing -Werror -DBOOST_THREAD_DONT_PROVIDE_PROMISE_LAZY" )
-SET(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -fno-strict-aliasing -Werror -DBOOST_THREAD_DONT_PROVIDE_PROMISE_LAZY" )
+SET(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -fno-strict-aliasing -Werror -DBOOST_THREAD_DONT_PROVIDE_PROMISE_LAZY -fvisibility=hidden" )
+SET(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -fno-strict-aliasing -Werror -DBOOST_THREAD_DONT_PROVIDE_PROMISE_LAZY -fvisibility=hidden" )
 
 # external_plugins needs to be compiled first because libraries/app depends on STEEM_EXTERNAL_PLUGINS being fully populated
 add_subdirectory( external_plugins )

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -42,7 +42,6 @@ LIST(APPEND BOOST_COMPONENTS thread
                              system
                              filesystem
                              program_options
-                             signals
                              serialization
                              chrono
                              unit_test_framework

--- a/libraries/fc/CMakeLists.txt
+++ b/libraries/fc/CMakeLists.txt
@@ -34,7 +34,7 @@ endif()
 SET (ORIGINAL_LIB_SUFFIXES ${CMAKE_FIND_LIBRARY_SUFFIXES})
 
 SET(BOOST_COMPONENTS)
-LIST(APPEND BOOST_COMPONENTS thread date_time system filesystem program_options signals serialization chrono unit_test_framework context locale iostreams)
+LIST(APPEND BOOST_COMPONENTS thread date_time system filesystem program_options serialization chrono unit_test_framework context locale iostreams)
 SET( Boost_USE_STATIC_LIBS ON CACHE STRING "ON or OFF" )
 
 IF( ECC_IMPL STREQUAL openssl )

--- a/libraries/fc/src/network/gntp.cpp
+++ b/libraries/fc/src/network/gntp.cpp
@@ -115,7 +115,6 @@ namespace fc
         {
           try 
           {
-            boost::system::error_code ec;
             ilog("Attempting to connect to GNTP srvice");
             asio::tcp::connect(*sock, eps[i]);
             endpoint = eps[i];

--- a/libraries/fc/src/thread/context.hpp
+++ b/libraries/fc/src/thread/context.hpp
@@ -1,8 +1,5 @@
 #pragma once
 #include <fc/thread/thread.hpp>
-#pragma GCC diagnostic push
-#pragma GCC diagnostic ignored "-Wunused-variable"
-#pragma GCC diagnostic pop
 #include <fc/exception/exception.hpp>
 #include <vector>
 
@@ -16,7 +13,10 @@
 #if BOOST_VERSION >= 106800
 #include <boost/context/continuation_fcontext.hpp>
 #else
+#pragma GCC diagnostic push
+#pragma GCC diagnostic ignored "-Wunused-variable"
 #include <boost/context/all.hpp>
+#pragma GCC diagnostic pop
 #endif
 
 #if BOOST_VERSION >= 106100

--- a/libraries/fc/src/thread/context.hpp
+++ b/libraries/fc/src/thread/context.hpp
@@ -2,7 +2,6 @@
 #include <fc/thread/thread.hpp>
 #pragma GCC diagnostic push
 #pragma GCC diagnostic ignored "-Wunused-variable"
-#include <boost/context/all.hpp>
 #pragma GCC diagnostic pop
 #include <fc/exception/exception.hpp>
 #include <vector>
@@ -13,6 +12,12 @@
 
 #define BOOST_COROUTINES_NO_DEPRECATION_WARNING // Boost 1.61
 #define BOOST_COROUTINE_NO_DEPRECATION_WARNING // Boost 1.62
+
+#if BOOST_VERSION >= 106800
+#include <boost/context/continuation_fcontext.hpp>
+#else
+#include <boost/context/all.hpp>
+#endif
 
 #if BOOST_VERSION >= 106100
   #include <boost/coroutine/stack_allocator.hpp>

